### PR TITLE
Break JSX element if a string literal prop has a newline

### LINKED ISSUE #3803: "jsxBracketSameLine: false" doesn't work correctly if prop value spans across multiple lines
**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4D4A6UABEQHTmGoD0WeA3IZQIa7mnUsNSNQY5EBnAK4AHOACcAMtADmABSbimAWwC8+EACNFGokygQYACwkBlUROlR5ileq1MAXroCecGAEEDxsxamyFJTUNbWcQAl4aHFwuShEmYyJ0ewBZABZSAHYANgBWIgBGbNIADgAmAE4AGwBmUgAGbIqAWjLSCrTsolzSgprurLSSwhJRsYKC0hqa5onSMpqyojr6xuXSApLZjJqugoz6kvqwokIqGJAAGhAIERhMaAFkUBFxTFgAdUx0Y2Qj65gTE0Xx+RmQZWuAneMiqcAAikJDHBkAAzJhVARwAFKTBVaEAYQgymUTGQIAMUGR100SjAAGt3KZ4mBocgYOIhFiQAArAQADwAQrSGTBTCo4JJ3sikGiMVz4uJMeIydpNC4qtAriBMcpMGyOVyhJiACpAp4y9GY67vJUwORKGQk1GWrniOAARyEmDd9qYjtJFrl11eEExHyUIjJrzgSuwVJAinEEAA7oEEOaE9gIN8tegIGBnUGEwIYIWrTchDARJWymW4ABfetAA)

**Input:**
```jsx
<div>
  ...
</div>;

<a>...</a>;


<div superLongParam="bar" anotherSuperLongParam="baz" yetAnotherSuperLongParam="baz">
</div>;

<path d="M4.765 16.829l3.069-2.946 5.813 5.748
  11.33-11.232 3.006 3.18-14.36 14.080z" 
/>;
```

**Output:**
```jsx
<div>...</div>;

<a>...</a>;

<div
  superLongParam="bar"
  anotherSuperLongParam="baz"
  yetAnotherSuperLongParam="baz"
/>;

<path d="M4.765 16.829l3.069-2.946 5.813 5.748
  11.33-11.232 3.006 3.18-14.36 14.080z" />;
```

**Expected behavior:**
```jsx
<div>...</div>;

<a>...</a>;

<div
  superLongParam="bar"
  anotherSuperLongParam="baz"
  yetAnotherSuperLongParam="baz"
/>;

<path d="M4.765 16.829l3.069-2.946 5.813 5.748
  11.33-11.232 3.006 3.18-14.36 14.080z"
/>
```

`jsxBracketSameLine: false` option handles most cases correctly, but I'd expect the closing bracket in the last element (`path`) to be moved to a separate line. I'm using `eslint-config-airbnb` and its `'react/jsx-closing-bracket-location': ['error', 'line-aligned']` rule is violated after running this code through prettier.

### PR DESCRIPTION
**Pull request by duailibe**
Closes #3803

### CODE REVIEW

**Comment by duailibe** [comment]
Huh.. I've spent a couple of minutes trying to figure out why `/>` was not printing on the same line with `--jsx-bracket-same-line` only to discover that's intended:


**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAwgAgA4CcLYC8wwAOlJpZgIbmUC+9mA9AHwgA0IBMAltAGdkoarnwB3AAqiEQlNQBuEXgBNOIagJjIYuAK5wuAI1zUwAazgwAytjO8oAcx37DIFRDDIAZtQA2Am4OgbgwkqaOALbUPv6BXABWAgAeAEKmFlbW1JFwADIOcC4GXBB6MNjlAEyxAW52uCHIIEbURgCeftDqeA4wAOqqMAAWyAAcAAxceBCB-abYzXhwIQpFXLhwAI56vJvh1FExSL51XIGRvMVuAg6OfnAAinoQ8LXxIDBtgyojyFVcXTUXh+O7oCCRaLNKDQdYgPSBAAqbTkp0CjCAA)
```sh
--jsx-bracket-same-line
```

**Input:**
```jsx
<C prop={{
    a
  }} />
```

**Output:**
```jsx
<C
  prop={{
    a
  }}
/>;

```

I've never noticed it before

**Comment by duailibe** [comment]
You might want to review it again, it didn't fix all the cases before :)

File: tests/jsx_last_line/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -48,3 +48,48 @@ exports[`last_line.js 2`] = `
...
+  11.33-11.232 3.006 3.18-14.36 14.080z" />;
+
+<Component text="Text
+  with
+  newlines">Content</Component>;
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<path
+  d="M4.765 16.829l3.069-2.946 5.813 5.748
+  11.33-11.232 3.006 3.18-14.36 14.080z"
+/>;
```
Comment (j-f1) [inline tests/jsx_last_line/__snapshots__/jsfmt.spec.js.snap]: Shouldn’t this end up on the same line since `jsxBracketSameLine` is `true`?

Comment (duailibe) [reply tests/jsx_last_line/__snapshots__/jsfmt.spec.js.snap]: Nope

**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAwgHSgAhwBwCcJ8BBAXkxABtoBzWqB+x5p1jl+q7PIkgEKUaXdqLYTOYpjygB6AHwBuEABoQJGAEtoAZ2SgAhoWIB3AArGE+lIYBuELQBM1IQ7pjIYhAK5x1AEaEhmAA1nAwAMr4IVpMXr7+IE4QYMgAZobUuklxOYQw5sF0ALaGGVk56gBWugAeAsFhEZGGJXAAMnFwCX7qED4w+IMATBXZSTGE+cggAYYBAJ6MrkRxMADqzjAAFsgAHAAM6vw5G8H4s0Rw+XY96oRwAI4+Wo9FhqXlSJkT6jklLS9JK6OIMOAARR8EHg4yqIBgCy2Tl2yBG6m8hi01DB6AgJTKsyg0HuIB8OQAKgsbL8cgBfOlAA)
```sh
--jsx-bracket-same-line
```

**Input:**
```jsx
<C
  propA="longlonglonglonglonglonglonglonglong"
  propB="longlonglonglonglonglonglonglonglong"
/>;
```

**Output:**
```jsx
<C
  propA="longlonglonglonglonglonglonglonglong"
  propB="longlonglonglonglonglonglonglonglong"
/>;

```

Comment (j-f1) [reply tests/jsx_last_line/__snapshots__/jsfmt.spec.js.snap]: Oh OK so it’s only inlined if it’s an opening element.

**Pull request merged**